qiskit/circuit/__init__.py

```diff
@@ -54,81 +78,825 @@
    circ.draw('mpl')
 
 
-Supplementary Information
-=========================
+.. _circuit-definitions:
+
+Circuit concepts and definitions
+================================
+
+There is a lot of specialized terminology around quantum circuits.  Much of this is common in
+quantum-computing literature, while some is more specific to quantum software packages, and a small
+amount specific to Qiskit.  This is an alphabetical listing of some of the important concepts as a
+quick reference, but does not go into detail of the foundational concepts.  Consider using the `IBM
+Quantum Learning platform <https://learning.quantum.ibm.com/>`_ if you want to start from the
+beginning.
+
+abstract circuit
+    A *circuit* defined in terms of abstract mathematical operations and *virtual qubits*.  This is
+    typically how you think about quantum algorithms; an abstract circuit can be made up of
+    completely arbitrary unitary operations, measurements, and potentially *near-time classical
+    computation*, with no restrictions about which qubits can interact with each other.
+
+    You turn an abstract circuit into a *physical circuit* by using :ref:`Qiskit's transpilation
+    package <qiskit-transpiler>`.
+
+circuit
+    A computational routine the defines a single execution to be taken on a QPU.  This can either be
+    an *abstract circuit* or a *physical circuit*.
+
+gate
+    A *unitary operation* on one or more qubits.
+
+hardware qubit
+    The representation of a single qubit on a particular *QPU*.  A hardware qubit has some physical
+    quantum-mechanical system backing it; unlike a *virtual qubit*, it has particular coupling
+    constraints and only certain gates can be applied to certain physical qubits.
+
+    Qiskit does not distinguish *logical qubits* from any individual *physical qubits* when talking
+    about hardware qubits.  A QPU may implement its hardware qubits as logical qubits, where each
+    hardware qubit comprises many physical qubits that are controlled and error-corrected opaquely
+    to Qiskit by the control electronics.  More likely, for near-term applications, a QPU will be
+    directly exposing its physical qubits as the hardware qubits for Qiskit to reason about.
+
+    Both physical and logical qubits will have coupling constraints between them, only permit
+    certain quantum operations on them, and have scheduling concerns between them.  Qiskit abstracts
+    these concerns together in the concept of hardware qubits. In the early days of quantum error
+    correction, particular backends may let you access their qubit resources either as high-level
+    logical qubits or as low-level physical qubits through Qiskit.
```

Comment:
I think we should not talk about logical qubits at all.

Reply:
(I'll just close this one in favour of https://github.com/Qiskit/qiskit/pull/11904#discussion_r1506721949 to avoid duplication, since we'll need to go through more than just these two parts pending the resolution of that anyway.)